mermin/src/span/direction.rs

```diff
@@ -160,27 +163,63 @@ impl DirectionInferrer {
             port: flow_key.dst_port,
             protocol: stats.protocol,
         };
-        if map.get(&dst_key, 0).is_ok() {
-            return Some(ClientServer {
-                client_ip: *src_ip,
-                client_port: flow_key.src_port,
-                server_ip: *dst_ip,
-                server_port: flow_key.dst_port,
-            });
+        match map.get(&dst_key, 0) {
+            Ok(_) => {
+                inc_ebpf_map_ops(
+                    EbpfMapName::ListeningPorts,
+                    EbpfMapOperation::Read,
+                    EbpfMapStatus::Ok,
+                );
+                return Some(ClientServer {
+                    client_ip: *src_ip,
+                    client_port: flow_key.src_port,
+                    server_ip: *dst_ip,
+                    server_port: flow_key.dst_port,
+                });
+            }
+            Err(e) => {
+                // Track not_found (expected) vs error (unexpected)
+                let status = if e.to_string().contains("not found")
+                    || e.to_string().contains("No such file")
+                {
+                    EbpfMapStatus::NotFound
+                } else {
+                    EbpfMapStatus::Error
+                };
+                inc_ebpf_map_ops(EbpfMapName::ListeningPorts, EbpfMapOperation::Read, status);
+            }
```

Comment:
Is there a way to do this without converting to string? I find it hard to imagine you couldn't compare types or enums here.